src/main/java/com/fasterxml/jackson/databind/DeserializationContext.java

```diff
@@ -206,7 +206,7 @@ protected DeserializationContext(DeserializationContext src,
             DeserializationConfig config, JsonParser p,
             InjectableValues injectableValues)
     {
-        _cache = src._cache;
+        _cache = src._cache.withCache(config.getCacheProvider());
```

Comment:
Hmmh. Not sure what this is supposed to do, but ideally no more copies should be made here?
(I may need to have a look at code why this is necessary).

Reply:
Ok, no, this configuration of `DeserializerCache` needs to occur earlier and not here: when `DeserializerCache` is added to `DeserializationConfig`. Note how `_cache` was used as-is earlier -- that should remain to be the case.

I think this may need to be done when setting `CacheProvider` for `ObjectMapper`.